### PR TITLE
Fix freeze after closing terminal on windows

### LINKED ISSUE #4430: 5.3.0-latest.2021125.2 crash frequently on win10
windows version:  win10 20H2 19042.985

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes https://github.com/lensapp/lens/issues/4430

### CODE REVIEW

File: src/renderer/api/terminal-api.ts [hunk 1]
```diff
@@ -144,14 +144,6 @@ export class TerminalApi extends WebSocketApi<TerminalEvents> {
     this.socket.binaryType = "arraybuffer";
   }
 
-  destroy() {
```
Comment (Nokel81) [inline src/renderer/api/terminal-api.ts]: This is not needed, we kill the underlying process on normal socket close anyway

File: src/main/shell-session/shell-session.ts [hunk 2]
```diff
@@ -158,6 +158,8 @@ export abstract class ShellSession {
         cwd,
         env,
         name: "xterm-256color",
+        // TODO: Something else is broken here so we need to force the use of winPty on windows
+        useConpty: false,
```
Comment (jim-docker) [inline src/main/shell-session/shell-session.ts:162]: what does this do on mac and linux?

File: src/renderer/api/terminal-api.ts (hunk 1 above)
Comment (jim-docker) [reply src/renderer/api/terminal-api.ts]: Are you 100% sure this won't affect behaviour on all platforms? If this isn't related to the fix I'd leave it in.

File: src/main/shell-session/shell-session.ts (hunk 2 above)
Comment (Nokel81) [reply src/main/shell-session/shell-session.ts:162]: Nothing, this option is ignored on non-windows OS's

File: src/renderer/api/terminal-api.ts (hunk 1 above)
Comment (Nokel81) [reply src/renderer/api/terminal-api.ts]: Okay, I'll make a different PR to remove this.

File: src/main/shell-session/shell-session.ts (hunk 2 above)
Comment (jim-docker) [reply src/main/shell-session/shell-session.ts:162]: mention or reference the issue this is a workaround for, in the comment?

**Review by jim-docker — APPROVED** [review]
- The 1005 code issue persists, and the TypeError (no `url` on undefined). But no hang
- Console Window Host and winpty-agent.exe processes are getting orphaned (open a node shell, run `ls`, close the tab) Maybe this behaviour already existed.

- tested on windows
- lightly tested on mac and linux to make sure there are no side effects

If we need a fix now this looks like it will do.

**Review by jakolehm — APPROVED** [review]

**Pull request merged**